doc/source/data/examples/batch_inference_object_detection.ipynb

```diff
@@ -0,0 +1,350 @@
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from typing import Dict\n",
+    "\n",
+    "import numpy as np\n",
+    "from torchvision import transforms\n",
+    "from torchvision.models.detection import (FasterRCNN_ResNet50_FPN_V2_Weights,\n",
+    "                                          fasterrcnn_resnet50_fpn_v2)\n",
+    "\n",
+    "\n",
+    "class ObjectDetectionModel:\n",
+    "    def __init__(self):\n",
+    "        # Define the model loading and initialization code in `__init__`\n",
+    "        self.weights = FasterRCNN_ResNet50_FPN_V2_Weights.DEFAULT\n",
+    "        self.model = fasterrcnn_resnet50_fpn_v2(\n",
+    "            weights=self.weights,\n",
+    "            box_score_thresh=0.9,\n",
+    "        )\n",
+    "        self.model.eval()\n",
+    "        # Note, since the data in the batch input is `np.ndarray`s, \n",
+    "        # we need `transforms.ToTensor` to convert the data to torch tensors.\n",
+    "        self.preprossor = transforms.Compose(\n",
+    "            [transforms.ToTensor(), self.weights.transforms()]\n",
+    "        )\n",
+    "\n",
+    "    def __call__(self, batch: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:\n",
+    "        # Define the per-batch inference code in `__call__`\n",
+    "        # Preprocess the images.\n",
+    "        batch = [self.preprossor(image) for image in batch[\"image\"]]\n",
+    "        # Do inference on the images.\n",
+    "        predictions = self.model(batch)\n",
+    "        # Get the inferred labels and convert it to a np.ndarray\n",
+    "        labels = np.array(\n",
+    "            [\n",
+    "                \",\".join(\n",
+    "                    [self.weights.meta[\"categories\"][i] for i in prediction[\"labels\"]]\n",
+    "                )\n",
+    "                for prediction in predictions\n",
+    "            ],\n",
+    "            dtype=\"S\",\n",
+    "        )\n",
+    "        # `__call__` also returns a `Dict[str, np.ndarray]`.\n",
+    "        return {\"labels\": labels}"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then, we use the `map_batches` API to apply the model to the whole image data set. Here, the `compute` argument indicates the number of concurrent Ray actors that run the model, and the `batch_size` argument indicates the number of images in each batch."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ds = ds.map_batches(\n",
+    "    ObjectDetectionModel,\n",
+    "    compute=ray.data.ActorPoolStrategy(size=4),\n",
+    "    batch_size=4,\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Lastly, we can use `ds.show` to inspect samples in the result, or the the `ds.write_xxx` APIS to write results to external storages."
```

Comment:
add a step at the end that does `ds.take()` and can visualize the bounding boxes?